packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -552,6 +576,14 @@ export async function toAbsoluteGlob(
   }
   root = globSafePath(root)
   const dir = importer ? globSafePath(dirname(importer)) : root
+  if (base) {
+    if (base.startsWith('./')) return pre + posix.join(dir, base, glob)
+    if (glob[0] === '@') {
+      const withoutAlias = /\/.+$/.exec(glob)?.[0]
+      return pre + posix.join(root, base, withoutAlias || glob)
+    }
+  }
+  glob = base ? posix.join(base, glob) : glob
```

Comment:
Since `base` is now where we resolve globs from, I think what we should do instead is to update the `dir` here. For example, `base` should be used as `dir` if it's defined, otherwise we falllback to the existing logic. Like:

```js
let dir
if (base) {
  if (base.startsWith('/')) { // root-relative
    dir = posix.join(root, base)
  } else { // assume relative path from the current file
    dir = posix.resolve(importer, base)
  }
} else {
  dir = importer ? globSafePath(dirname(importer)) : root
}
```

We shouldn't need to update or return the glob ourselves. That would be like prefixing.

This assumes that `base` can only be `./foo` (relative to the current file) or `/src/foo` (relative from the root), which I think should be sufficient for an initial implementation. We might want to validate in `parseGlobOption` that it can only start with `./`, `../`. or `/`.

Reply:
Thanks, I've made the changes below.
https://github.com/vitejs/vite/pull/18510/commits/f6c657d0275fc0299b142d0a8f682b1c09d4c82f